File: 프로그래머스/2/42842. 카펫/카펫.js
function solution(brown, yellow) {
    //옐로우의 가로=옐로우 전체 개수/옐로우의 세로
    //브라운의 개수=옐로우의 세로*2+옐로우의 가로*2+4
    for(let i=1; i<=yellow; i++){
        if(yellow%i===0&&brown===i*2+yellow/i*2+4){
            return [yellow/i+2,i+2]
        }
    }
}

File: 프로그래머스/2/42842. 카펫/카펫.java
class Solution {
    public int[] solution(int brown, int yellow) {
        int yellow_height=0;
        int yellow_width=0;
        for(int i=1; i<=yellow; i++){
            if(yellow%i==0){ // i는 세로 길이 후보
                int height=i;
                int width=yellow/i;
                // System.out.println(height+" "+ width);
                if((width+2)*2+height*2==brown){
                    yellow_height=height;
                    yellow_width=width;
                    break;
                }
            }
        }
        
        return new int[]{yellow_width+2, yellow_height+2};
    }
}